File: car-stack.ts
/*
 * Class that defines string stack.
 *
 * @author  Andi Cucka
 * @version 1.0
 * @since   2024-04-25
 */

export class CarStack {
  private strStack: string[]

  // variables
  constructor() {
    self.strStack = []
  }

  // show full stack separated by commas
  public get show() {
    let values = ""
    for (let counter = 0; counter < self.strStack.length; counter++) {
      values = values + self.strStack[counter] + ", "
    }
    values = values.substring(0, values.length-2)
    return values
  }

  // push item to end of list
  public push(input) {
    self.strStack.push(input)
  }

  // pop item from end of list and return value
  public pop() {
    let removed = ""
    // checks if there's at least 1 element in the stack
    if (self.strStack.length > 0) {
      removed = self.strStack[self.strStack.length - 1]
    } else {
      removed = "nothing to remove"
    }
    self.strStack.splice(-1)
    return removed
  }
}

File: Car stack.java
/*
* The CarStack Class.
*
* @author  Andi Cucka
* @version 1.0
* @since   2024-03-27
*/

import java.util.ArrayList;
import java.util.List;

/**
* The CarStack() class.
*/
public class CarStack {
    /**
    * Declares a private instance variable.
    */
    private List<String> strStack = new ArrayList<>();

    /**
    * The push() method.
    *
    * @param input from Main.java
    */
    public void push(String input) {
        strStack.add(input);
    }

    /**
    * The showStack() method.
    *
    * @return the values as a string
    */
    public String showStack() {
        String values = "";
        for (int counter = 0; counter < strStack.size(); counter++) {
            values += strStack.get(counter);
            if (counter < strStack.size() - 1) {
                values += ", ";
            }
        }
    return values;
    }

    /**
    * This method pops a string from the stack.
    *
    * @return the popped string
    */
    public String popItem() {
        final String poppedItem;
        if (!strStack.isEmpty()) {
            poppedItem = strStack.remove(strStack.size() - 1);
        } else {
            poppedItem = null;
        }
        return poppedItem;
    }
}
